staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go

```diff
@@ -349,3 +350,12 @@ type StorageVersionProvider interface {
 	// list of kinds the object might belong to.
 	StorageVersion() runtime.GroupVersioner
 }
+
+// ResetFields maps versions to the sets of fields that will be reset by the storage strategy.
+type ResetFields map[fieldpath.APIVersion]*fieldpath.Set
+
+// ResetFieldsProvider is an optional interface that a storage object can
+// implement if it wishes to provide the fields reset by its strategies.
+type ResetFieldsProvider interface {
```

Comment:
Curious why you called that "provider" rather than "strategy"? Specifically since you call it "Strategy" in the variables somewhere else?